### PR TITLE
GP and SP standard atmosphere models

### PR DESCRIPTION
**Pull request by pgkirsch**
Creates models for air density as a function of altitude using a standard atmosphere model (valid up to the top of the troposphere). The GP model takes upward pressure on rho and altitude, and the SP model takes downward pressure on rho.

### CODE REVIEW

**Comment by bqpd** [comment]
(reviewed)

File: standard_atmosphere_SP.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        L   = Variable('L', L, 'K/m')        # Temperature lapse rate
+        MoR = Variable('MoR', M/R, 'kg*K/J') # M/R (see above)
+        p0  = Variable('p0', 101325, 'Pa')   # Sea level standard pressure
+        T0  = Variable('T0', 288.15, 'K')    # Sea level standard temperature
+
+        with SignomialsEnabled():
+            objective = rho
+            constraints = [
+                            # Model only valid up to top of the troposphere
+                            h <= 20000,#*units.m,
```
Comment (bqpd) [inline standard_atmosphere_SP.py]: `*units.m` should still work; when units are disabled it just returns a `1`

File: standard_atmosphere.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        # Free variables
+        h   = Variable('h', 'm')             # Altitude
+        rho = Variable('rho', 'kg/m^3')      # Air density
+        T   = Variable('T', 'K')             # Temperature
+
+        # Constants
+        L   = Variable('L', L, 'K/m')        # Temperature lapse rate
+        MoR = Variable('MoR', M/R, 'kg*K/J') # M/R (see above)
+        p0  = Variable('p0', 101325, 'Pa')   # Sea level standard pressure
```
Comment (bqpd) [inline standard_atmosphere.py]: rename `p_{sealevel}`?

File: standard_atmosphere.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    def setup(self):
+        g  = 9.81        # Acceleration due to gravity [m/s^2]
+        L  = 0.0065      # Temperature lapse rate [K/m]
+        M  = 0.0289644   # Molar mass of dry air [kg/mol]
+        R  = 8.31447     # Universal gas constant [J/(mol*K)]
+        th = (g*M)/(R*L) # [-]
+
+        # Free variables
+        h   = Variable('h', 'm')             # Altitude
+        rho = Variable('rho', 'kg/m^3')      # Air density
```
Comment (bqpd) [inline standard_atmosphere.py]: rename `"\\rho"`?

File: standard_atmosphere.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    Assumptions: only valid to the top of the troposphere (~20km)
+
+    source: https://en.wikipedia.org/wiki/Density_of_air#Altitude
+    """
+    def setup(self):
+        g  = 9.81        # Acceleration due to gravity [m/s^2]
+        L  = 0.0065      # Temperature lapse rate [K/m]
+        M  = 0.0289644   # Molar mass of dry air [kg/mol]
+        R  = 8.31447     # Universal gas constant [J/(mol*K)]
+        th = (g*M)/(R*L) # [-]
```
Comment (bqpd) [inline standard_atmosphere.py]: Make this (and the rest) properties of the model like `Beam`

**Comment by bqpd** [comment]
I put all my comments above into #10

File: standard_atmosphere.py (hunk 3 above)
Comment (whoburg) [inline standard_atmosphere.py]: or alternatively `r"\rho"` if you don't like double backslash

**Comment by whoburg** [comment]
This is looking great. Shall we merge #10 into it and then merge it into master?

**Comment by bqpd** [comment]
Merged the PR. Ready to adopt this model?

**Comment by whoburg** [comment]
This looks great, and I'm overall in favor of merging, with one idea to discuss first. 

One idea, which could be put into a separate PR to enable merging this now:
It's a bit of code duplication to have both `Troposphere` and `TroposphereSP`. Would it make any sense to have `TroposphereSP` inherit from `Model`, and then have either a method or subclass implement the GP version by replacing/approximating some Signomial constraints with Posynomials?

**Comment by bqpd** [comment]
I was initially in favour of that as well, but @pgkirsch noted that the two models have very different pressures, which seems like an excellent reason to keep them separate

**Comment by bqpd** [comment]
@whoburg, I'm ready to merge this?

**Pull request merged**